sqlalchemy_ibmi/base.py

```diff
@@ -772,13 +772,20 @@ def create_connect_args(self, url):
         opts = url.translate_connect_args(username='user', host='system')
         opts.update(url.query)
         allowed_opts = {'system', 'user', 'password',
-                        'autocommit', 'readonly', 'timeout', 'database'}
+                        'autocommit', 'readonly', 'timeout',
+                        'database', 'library_list', 'current_schema'}
         if allowed_opts < opts.keys():
             raise ValueError("Option entered not valid for "
                              "IBM i Access ODBC Driver")
- 
-        return [["Driver={%s}; UNICODESQL=1; TRUEAUTOCOMMIT=1" % (
-                 self.pyodbc_driver_name)], opts]
+
+        connection_string = "Driver={%s}; UNICODESQL=1; TRUEAUTOCOMMIT=1;" % (
+                 self.pyodbc_driver_name)
+        if 'current_schema' in opts or 'library_list' in opts:
+            libl = 'DefaultLibraries={},'.format(opts.get('current_schema', ''))
+            libl += ','.join(opts.get('library_list', ''))
+            connection_string = connection_string + libl + ';'
+
+        return [[connection_string], opts]
```

Comment:
So, I forgot about this before, but opts gets passed as the kwargs to pyodbc.connect, this means that without removing library_list and current_schema from opts, it will be passed to the driver _again_.

I remember now that we were talking about only setting things on the connection string that were _always_ going to be set a specific way (ie. not user configurable). We can pass user-configurable connection options in the `opts` parameter which will get passed to pyodbc, which will use it to generate a connection string to pass to the driver.

So, I think we should `pop()` the current_schema and library_list values from `opts` and then if one or more is set, build a DefaultLibraries key that is set on opts.